packages/amplify-category-function/src/provider-utils/awscloudformation/utils/cloudformationHelpers.ts

```diff
@@ -26,9 +26,24 @@ export function getNewCFNEnvVariables(oldCFNEnvVariables, currentDefaults, newCF
       deletedResources.push(resourceName);
     }
   });
+  const appsyncTableSuffix = '@model(appsync)';
+
+  const deleteAppSyncTableResources = deletedResources.filter(resource => resource.includes(appsyncTableSuffix.toUpperCase()));
```

Comment:
there's some similar logic to convert models to the corresponding storage resource name in execPermissionsWalkthrough.ts. maybe look through that and see if there's anything that can be pulled into shared functions?